docs/extended_api/pipelines/make_pipeline.md

```diff
@@ -0,0 +1,135 @@
+---
+grand_parent: Extended API
+parent: Pipelines
+---
+
+# `cuda::make_pipeline`
+
+```c++
+// Header `<cuda/pipeline>`.
+
+__host__ __device__
+cuda::pipeline<cuda::thread_scope_thread> cuda::make_pipeline();
+```
+
+Creates a _unified pipeline_ such that the calling thread is the only
+  participating thread and performs both producer and consumer actions.
+
+```c++
+// Header `<cuda/pipeline>`.
+
+template <typename Group,
+          cuda::thread_scope Scope, cuda::std::uint8_t StagesCount>
+__host__ __device__
+cuda::pipeline<Scope>
+cuda::make_pipeline(Group const& group,
+                    cuda::pipeline_shared_state<Scope, StagesCount>* shared_state);
+```
+
+Creates a _unified pipeline_ such that all the threads in `group` are
+  performing both producer and consumer actions.
+
+```c++
+// Header `<cuda/pipeline>`.
+
+template <typename Group,
+          cuda::thread_scope Scope, cuda::std::uint8_t StagesCount>
+__host__ __device__
+cuda::pipeline<Scope>
+cuda::make_pipeline(Group const& group,
+                    cuda::pipeline_shared_state<Scope, StagesCount>* shared_state,
+                    cuda::std::size_t producer_count);
+```
+
+Creates a _partitioned pipeline_ such that `producer_threads` number of threads
+  in `group` are performing producer actions while the others are performing
+  consumer actions.
+
+```c++
+// Header <cuda/pipeline>.
+
+template <typename Group,
+          cuda::thread_scope Scope, cuda::std::uint8_t StagesCount>
+__host__ __device__
+cuda::pipeline<Scope>
+cuda::make_pipeline(Group const& group,
+                    cuda::pipeline_shared_state<Scope, StagesCount>* shared_state,
+                    cuda::pipeline_role role);
+```
+
+Creates a _partitioned pipeline_ where each thread's role is explicitly
+  specified.
```

Comment:
Same comment as on the `memcpy_async` file; this feels much less readable than the previously used cppreference style.